standard_test.go

```diff
@@ -0,0 +1,85 @@
+package retryablehttp
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestStandardClient_Do(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
```

Comment:
You've used `http.MethodPost` in a few other places, would be nice to shore this up with that.